src/util/evalUnits.ts

```diff
@@ -13,33 +13,43 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import BN from 'bn.js';
+
+const getSiValue = (si: number): BN => new BN(10).pow(new BN(si));
+
 const si = [
-  { value: -1e24, symbol: 'y' },
-  { value: -1e21, symbol: 'z' },
-  { value: -1e18, symbol: 'a' },
-  { value: -1e15, symbol: 'f' },
-  { value: -1e12, symbol: 'p' },
-  { value: -1e9, symbol: 'n' },
-  { value: -1e6, symbol: 'μ' },
-  { value: -1e3, symbol: 'm' },
-  { value: 1, symbol: '' },
-  { value: 1e3, symbol: 'k' },
-  { value: 1e6, symbol: 'M' },
-  { value: 1e9, symbol: 'G' },
-  { value: 1e12, symbol: 'T' },
-  { value: 1e15, symbol: 'P' },
-  { value: 1e18, symbol: 'E' },
-  { value: 1e21, symbol: 'Z' },
-  { value: 1e24, symbol: 'Y' }
+  { value: getSiValue(24), symbol: 'y' },
+  { value: getSiValue(21), symbol: 'z' },
+  { value: getSiValue(18), symbol: 'a' },
+  { value: getSiValue(15), symbol: 'f' },
+  { value: getSiValue(12), symbol: 'p' },
+  { value: getSiValue(9), symbol: 'n' },
+  { value: getSiValue(6), symbol: 'μ' },
+  { value: getSiValue(3), symbol: 'm' },
+  { value: new BN(1), symbol: '' },
+  { value: getSiValue(3), symbol: 'k' },
+  { value: getSiValue(6), symbol: 'M' },
+  { value: getSiValue(9), symbol: 'G' },
+  { value: getSiValue(12), symbol: 'T' },
+  { value: getSiValue(15), symbol: 'P' },
+  { value: getSiValue(18), symbol: 'E' },
+  { value: getSiValue(21), symbol: 'Y' },
+  { value: getSiValue(24), symbol: 'Z' }
 ];
 
-const floats = /^[0-9]*[.,]{1}[0-9]*$/;
-const ints = /^[0-9]+$/;
-const alphaFloats = /^[0-9]*[.,]{1}[0-9]*[a-zA-Z]{1}$/;
-const alphaInts = /^[0-9]*[a-zA-Z]{1}$/;
+const allowedSymbols = si
+  .map((s) => s.symbol)
+  .join(', ')
+  .replace(', ,', ',');
+const floats = '^[+]?[0-9]*[.,]{1}[0-9]*$';
+const ints = '^[+]?[0-9]+$';
+const alphaFloats = '^[+]?[0-9]*[.,]{1}[0-9]*[' + allowedSymbols + ']{1}$';
+const alphaInts = '^[+]?[0-9]*[' + allowedSymbols + ']{1}$';
 
 export enum EvalMessages {
   GIBBERISH = 'Input is not correct. Use numbers, floats or expression (e.g. 1k, 1.3m)',
+  ZERO = 'You cannot send 0 funds',
```

Comment:
Technically you can yes - but is there a scenario that we should allow user to send 0 funds? should I remove this check?

Reply:
No, I was just commenting on the quite "strong" statement that you are making here. It's probably possible to do a PHD on displaying error messages to the user but I usually tend to be more in favour of soft, positive and helpful language instead of being strict and technical. I don't suggest any change, cause it would be out of context for this one PR - we tend to have very strict and technical error messages anyway, but for instance I'd be for something more like:
```js
GIBBERISH = "We can't understand the amount. Try typing just numbers with unit prefixes (e.g. 1k, 1.3m, etc)"
ZERO = "Seems you are trying to send 0, please check the amount."
SYMBOL_ERROR = "We don't know the symbol '{}', try using one of '{}'."
GENERAL_ERROR = "You've run into an unusual situation that we didn't handle correctly. Please consider reporting this as a bug if you think it should work". // This one is actually not used in the new code, so we should get rid of it (or can do it in `try-catch`).
```